src/Microsoft.Restier.EntityFramework/Submit/ChangeSetPreparer.cs

```diff
@@ -209,6 +209,24 @@ private static object ConvertIfNecessary(Type type, object value)
                 return (DateTime)dateValue;
             }
 
+            // Convert to System.DateTime supported by EF from DateTimeOffset.
+            if (value is DateTimeOffset)
```

Comment:
We have almost identical code here:) I just need to move some of my comments here later after your commit.